segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -202,14 +211,15 @@ private void logEnd() {
                                 chunkedSegmentStorage.getReadIndexCache().addIndexEntry(handle.getSegmentName(), currentChunkName, startOffsetForCurrentChunk.get());
                             }
                             cntScanned.incrementAndGet();
-                        }, chunkedSegmentStorage.getExecutor())
-                        .thenAcceptAsync(v -> {
-                            val elapsed = readIndexTimer.getElapsed();
-                            SLTS_READ_INDEX_SCAN_LATENCY.reportSuccessEvent(elapsed);
-                            log.debug("{} read - chunk lookup - segment={}, offset={}, scanned={}, latency={}.",
-                                    chunkedSegmentStorage.getLogPrefix(), handle.getSegmentName(), offset, cntScanned.get(), elapsed.toMillis());
                         }, chunkedSegmentStorage.getExecutor()),
-                chunkedSegmentStorage.getExecutor());
+                chunkedSegmentStorage.getExecutor())
+                .thenAcceptAsync(v -> {
```

Comment:
Why async? Those operations should be quick. No need to add a new task in the executor queue.

Reply:
The java documentation is ambiguous about it whether it is guaranteed to be run on same thread.
It seems it can run on this thread pool or thread pool of the caller. (in this case that would be possibly core thread).
Hence to be careful not to hijack core threads I choose to explicitly provide thread pool I want this to be run on. (which is same thread pool anyway).

Here I'm just maintaining consistency with rest of the code in this file by using Async. 
I do plan to use non async overload if it can be guaranteed that FJ or other thread pools don't get used accidently or a more likely case when storage thread pool is super busy.